instance-scheduler/Dockerfile

```diff
@@ -1,16 +1,26 @@
+# Stage 1: Build the Go binary
 FROM public.ecr.aws/lambda/go:1 as builder
 
-# Copy function code
-COPY . .
+# Install the specific version of Go
+ARG GO_VERSION=1.23.0
```

Comment:
Considering that this image is being deprecated, it may be worth using one of the [official Go Docker images](https://hub.docker.com/_/golang) for the build process 🐳 It'd save us having to manually install Go - and dependabot should be able to raise PRs to bump the docker image version, which will bump the Go version 🥳